rfcs/react-components/styles-handbook.md

```diff
@@ -0,0 +1,921 @@
+      '0%': { right: '100% /* @noflip */' },
+    },
+  },
+});
+```
+
+In this case automatic flipping is disabled and will produce less CSS: 2 classes instead of 4.
+
+## Nested selectors
+
+Use nested selectors responsibly in styles because they can cause "CSS rule explosion".
+
+### Use nested selectors with pseudo classes
+
+Use nested selectors when they are combined with pseudo classes:
+
+```js
+import { makeResetStyles, mergeClasses, shorthands } from '@griffel/react';
+import { tokens } from '@fluentui/react-theme';
+
+const useBaseClassName = makeResetStyles({
+  base: {
+    // ✅ Do
+    // Shows filled icon on hover
+    ':hover': {
+      [`& .${iconFilledClassName}`]: {
+        display: 'inline',
+      },
+      [`& .${iconRegularClassName}`]: {
+        display: 'none',
+      },
+    },
+  },
+});
+```
+
+### Apply classes directly to elements
+
+Do not use nested selectors to target an element or slot if you can apply classes on directly on it.
+
+```jsx
+import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
+import { tokens } from '@fluentui/react-theme';
+
+const useClasses = makeStyles({
+  slotA: {
+    backgroundColor: tokens.colorNeutralBackground1,
+    // ❌ Don't do
+    // You can apply classes directly to that "div"
+    '> div': {
+      color: tokens.colorNeutralForeground1,
+    },
+  },
+
+  // ✅ Do
+  slotA: {
+    backgroundColor: tokens.colorNeutralBackground1,
+  },
+  slotB: {
+    color: tokens.colorNeutralForeground1,
+  },
+});
+
+function App(props) {
+  const classes = useClasses();
+
+  return (
+    <div className={classes.root}>
+      {/* 💡 Classes can be passed directly to this element */}
+      <div className={classes.slot} />
+    </div>
+  );
+}
+```
+
+### Avoid complicated selectors
+
+Keep selectors simple to produce reusable CSS rules:
+
+- CSS rules that are unique cannot be reused in other areas
+  ```css
+  /*    ⬇️ cannot be reused in other components  */
+  .hash .some-unique-class {
+    display: flex;
+  }
+  ```
+- Complicated selectors produce bigger bundle size with AOT
+
+  ```js
+  makeStyles({
+    rootA: {
+      display: 'flex',
+    },
+    rootB: {
+      '> .some-classname': {
+        '> .other-classname': {
+          display: 'flex',
+          alignItems: 'center',
+        },
+      },
+    },
+  });
+  ```
+
+  ⬇️⬇️⬇️
+
+  ```css
+  /* ✅ no selectors */
+  .f22iagw {
+    display: flex;
+  }
+
+  /* ⚠️ with complex selectors */
+  .f1312jvm > .some-classname > .other-classname {
+    display: flex;
+  }
+  .f1c58nry > .some-classname > .other-classname {
+    align-items: center;
+  }
+  ```
+
+- Complicated selectors are hard to override as overrides should match component's styles
+
+  ```js
+  // On component side
+  makeResetStyles({
+    '> .some-classname': {
+      '> .other-classname': {
+        ':hover': {
+          display: 'flex',
+          alignItems: 'center',
+        },
+      },
+    },
+  });
+  // On consumer side 💥
```

Comment:
What does `// On consumer side 💥` mean? Does it mean:
`'> .some-classname > .other-classname:hover'` works but it's hard for consumer to guess it
or
`'> .some-classname > .other-classname:hover'` doesn't work because it doesn't match the structure on component side

Reply:
> works but it's hard for consumer to guess it

Great callout 👍 I updated the comment in d6a7704